sql_export/sql_export.py

```diff
@@ -0,0 +1,147 @@
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv.orm import Model
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+import StringIO
+import base64
+import datetime
+import re
+
+
+class SqlExport(Model):
+    _name = "sql.export"
+    _description = "SQL export"
+
+    PROHIBITED_WORDS = [
+        'delete',
+        'drop',
+        'insert',
+        'alter',
+        'truncate',
+        'execute',
+        'create',
+        'update'
+    ]
+
+    def _check_query_allowed(self, cr, uid, ids, context=None):
+        for obj in self.browse(cr, uid, ids, context=context):
+            query = obj.query.lower()
+            for word in self.PROHIBITED_WORDS:
+                expr = r'\b%s\b' % word
+                is_not_safe = re.search(expr, query)
+                if is_not_safe:
+                    return False
+        return True
+
+    def _get_editor_group(self, cr, uid, *args):
+        model_data_obj = self.pool['ir.model.data']
+        return [model_data_obj.get_object_reference(
+            cr, uid, 'sql_export', 'group_sql_request_editor')[1]]
+
+    _columns = {
+        'name': fields.char('Name', required=True),
+        'query': fields.text(
+            'Query',
+            required=True,
+            help="You can't use the following word : delete, drop, create, "
+                 "insert, alter, truncate, execute, update"),
+        'copy_options': fields.char(
+            'Copy Options',
+            required=True),
+        'group_ids': fields.many2many(
+            'res.groups',
+            'groups_sqlquery_rel',
+            'sql_id',
+            'group_id',
+            'Allowed Groups'),
+        'user_ids': fields.many2many(
+            'res.users',
+            'users_sqlquery_rel',
+            'sql_id',
+            'user_id',
+            'Allowed Users'),
+    }
+
+    _defaults = {
+        'group_ids': _get_editor_group,
+        'copy_options': "CSV HEADER DELIMITER ';'",
+    }
+
+    _constraints = [(_check_query_allowed,
+                    'The query you want make is not allowed : prohibited '
+                     'actions (delete, drop, insert, alter, truncate, '
+                     'execute, create, update)', ['query'])]
```

Comment:
Nitpick: maybe use 
                    'The query you want make is not allowed : prohibited '
                     'actions (%s)' % PROHIBITED_WORDS.join(', '),
                     ['query'])]